### PR TITLE
Sort diagnostics node dependencies so that order is stable

### LINKED ISSUE #4081: Inspector goldens are unstable
The list of `dependencies` here does not have a stable order, which causes spurious failures in our golden testing. 
Screen Shot 2022-05-06 at 3 22 08 PM

### PR DESCRIPTION
**Pull request by kenzieschmoll**
This will fix https://github.com/flutter/devtools/issues/4081.
@jacob314

### CODE REVIEW

File: packages/flutter/lib/src/widgets/framework.dart [hunk 1]
```diff
@@ -4455,7 +4455,8 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
     if (_dependencies != null && _dependencies!.isNotEmpty) {
       final List<DiagnosticsNode> diagnosticsDependencies = _dependencies!
         .map((InheritedElement element) => element.widget.toDiagnosticsNode(style: DiagnosticsTreeStyle.sparse))
-        .toList();
+        .toList()
+        ..sort((DiagnosticsNode a, DiagnosticsNode b) => '$a'.compareTo('$b'));
```
Comment (jacob314) [inline packages/flutter/lib/src/widgets/framework.dart:4459]: the toString for a DiagnosticsNode can be long and expensive (although shouldn't really be in this case). I would just use
toStringShort() instead to be safe toi make sure we don't end up doing O(n log n) computations of the full toString version of a diagnostic.

**Review by jacob314 — APPROVED** [review]
lgtm

File: packages/flutter/lib/src/widgets/framework.dart [hunk 2]
```diff
@@ -4455,7 +4455,12 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
     if (_dependencies != null && _dependencies!.isNotEmpty) {
       final List<DiagnosticsNode> diagnosticsDependencies = _dependencies!
         .map((InheritedElement element) => element.widget.toDiagnosticsNode(style: DiagnosticsTreeStyle.sparse))
```
Comment (jacob314) [inline packages/flutter/lib/src/widgets/framework.dart:4457]: Sort while you still have a List<Widget> and you can use toStringShort or just the runtime type of the Widget class which would also be reasonable.

Comment (kenzieschmoll) [reply packages/flutter/lib/src/widgets/framework.dart:4457]: good idea. done.

**Review by jacob314 — APPROVED** [review]
lgtm

**Pull request merged**